pkg/cluster/fetch.go

```diff
@@ -117,14 +117,14 @@ func BundlesForCluster(ctx context.Context, client Client, cluster *v1alpha1.Clu
 
 func eksaReleaseForCluster(ctx context.Context, client Client, cluster *v1alpha1.Cluster) (*v1alpha1release.EKSARelease, error) {
 	eksaRelease := &v1alpha1release.EKSARelease{}
-	if cluster.Spec.BundlesRef == nil {
-		if cluster.Spec.EksaVersion == nil {
-			return nil, fmt.Errorf("either cluster's EksaVersion or BundlesRef need to be set")
-		}
+	if cluster.Spec.EksaVersion == nil && cluster.Spec.BundlesRef == nil {
```

Comment:
This validation should be done at the caller of this function and not here, because this is purely for grabbing the eksa release. We should have this check in both `BundlesForCluster` and `BuildSpecFromConfig`